public/app/features/logs/logsModel.ts

```diff
@@ -611,15 +611,31 @@ type LogsVolumeQueryOptions<T extends DataQuery> = {
   range: TimeRange;
 };
 
+function defaultExtractLevel(dataFrame: DataFrame): LogLevel {
+  let valueField;
+  try {
+    valueField = new FieldCache(dataFrame).getFirstFieldOfType(FieldType.number);
+  } catch {}
+  return valueField?.labels ? getLogLevelFromLabels(valueField.labels) : LogLevel.unknown;
+}
+
+function getLogLevelFromLabels(labels: Labels): LogLevel {
+  const levelLabel = labels['level'] ?? labels['lvl'] ?? labels['loglevel'] ?? '';
+  return levelLabel ? getLogLevelFromKey(labels[levelLabel]) : LogLevel.unknown;
+}
+
 /**
  * Creates an observable, which makes requests to get logs volume and aggregates results.
  */
 export function queryLogsVolume<TQuery extends DataQuery, TOptions extends DataSourceJsonData>(
   datasource: DataSourceApi<TQuery, TOptions>,
   logsVolumeRequest: DataQueryRequest<TQuery>,
-  options: LogsVolumeQueryOptions<TQuery>
+  options?: LogsVolumeQueryOptions<TQuery>
```

Comment:
Yes, range and targets are required for re-using results on zoom-in: https://github.com/grafana/grafana/blob/aa670280fc46b2d96e3096f1395e5145c715cb44/public/app/features/explore/state/query.ts#L878

Reply:
@ifrost I really don't see where `range` and `targets` are used for zoom-in, could you please point me to where are they used?

Reply:
It took me so long to figure out why it was not working, because this not null assertion operator (!) was hiding a potential bug: https://github.com/grafana/grafana/blob/main/public/app/features/explore/utils/supplementaryQueries.ts#L82

Reply:
>  I don't see the point of using options.range. Same for options.targets.

`options.targets` was needed to pass the targets from the original requests not targets of the log volume requests. 

`options.range` was needed in the past when a partial log volume was rendered and the returned range didn't match the request range. But we moved the implementation of partial log volume out of data sources so maybe we don't need it anymore 🤔

Reply:
> options.targets was needed to pass the targets from the original requests not targets of the log volume requests.

Yeah! I eventually saw this, and updated the implementation to allow reuse of supplementary queries.

> But we moved the implementation of partial log volume out of data sources so maybe we don't need it anymore 🤔

It seems that we don't. `targets` was being modified inside `getDataProvider`, but range was always === to the original one.